common/src/main/java/com/discordsrv/common/feature/onlinesync/OnlineSyncModule.java

```diff
@@ -0,0 +1,225 @@
+        
+        if (player == null || player instanceof OfflinePlayer) {
+            return Task.failed(new SyncFail(OnlineSyncResult.PLAYER_NOT_ONLINE));
+        }
+
+        return Task.completed(!discordSRV.isShutdown());
+    }
+
+    @Override
+    protected Task<ISyncResult> applyDiscord(OnlineSyncConfig config, long userId, @Nullable Boolean newState) {
+        DiscordGuild guild = discordSRV.discordAPI().getGuildById(config.serverId);
+        if (guild == null) {
+            return Task.completed(GenericSyncResults.GUILD_NOT_FOUND);
+        }
+
+        DiscordRole role = guild.getRoleById(config.roleId);
+        if (role == null) {
+            return Task.completed(OnlineSyncResult.ROLE_DOESNT_EXIST);
+        }
+
+        if (!guild.getSelfMember().canInteract(role)) {
+            return Task.completed(OnlineSyncResult.ROLE_CANNOT_INTERACT);
+        }
+
+        return guild.retrieveMemberById(userId)
+                .then(member -> Boolean.TRUE.equals(newState) ? member.addRole(role).thenApply(v -> (ISyncResult) GenericSyncResults.ADD_DISCORD) : member.removeRole(role).thenApply(v -> GenericSyncResults.REMOVE_DISCORD))
+                .mapException(RestErrorResponseException.class, t -> {
+                    if (t.getErrorCode() == ErrorResponse.UNKNOWN_MEMBER.getCode()) {
+                        throw new SyncFail(OnlineSyncResult.NOT_A_GUILD_MEMBER);
+                    }
+                    if (t.getErrorCode() == ErrorResponse.MISSING_PERMISSIONS.getCode()) {
+                        throw new SyncFail(GenericSyncResults.MEMBER_CANNOT_INTERACT);
+                    }
+                    throw t;
+                });
+    }
+
+    @Override
+    protected Task<ISyncResult> applyGame(OnlineSyncConfig config, UUID playerUUID, @Nullable Boolean newState) {
+        return Task.completed(GenericSyncResults.WRONG_DIRECTION);
+    }
+
+    @Override
+    public void onPlayerConnected(PlayerConnectedEvent event) {
+        gameChanged(
+                OnlineSyncCause.PLAYER_JOINED_SERVER,
+                Someone.of(event.player().uniqueId()),
+                Game.INSTANCE,
+                true
+        );
+    }
+
+    @Subscribe
+    public void onPlayerDisconnected(PlayerDisconnectedEvent event) {
+        gameChanged(
+                OnlineSyncCause.PLAYER_LEFT_SERVER,
+                Someone.of(event.player().uniqueId()),
+                Game.INSTANCE,
+                false
+        );
+    }
+
+    @Subscribe
+    public void onShutdown(DiscordSRVShuttingDownEvent event) {
+        serverChanged(OnlineSyncCause.SERVER_SHUTDOWN, true);
+    }
+
+    @Override
+    public void reload(Consumer<ReloadResult> resultConsumer) {
+        super.reload(resultConsumer);
+        serverChanged(OnlineSyncCause.SERVER_STARTUP, false);
+    }
+
+    protected void serverChanged(OnlineSyncCause cause, boolean skipOnlineCheck) {
+        if (!discordSRV.config().onlineSync.isSet()) {
+            return;
+        }
```

Comment:
This isn't needed, the module will not be reloaded or be subscribed to the event bus if it isn't enabled (and `AbstractSyncModule#isEnabled` checks that at least one syncable is set)